### ISSUE TITLE
Use X-Plane CDU datarefs and upstream ExtPlane plugin

### ISSUE THREAD

**Issue by vranki** [waynepiekarski/XPlaneCDU#7]
X-Plane 11.35 introduced datarefs to access CDU display: https://developer.x-plane.com/article/datarefs-for-the-cdu-screen/

You could now use those datarefs to implement XPlaneCDU and use standard ExtPlane instead of forked one.

I've already implemented CDU display on ExtPlane-Panel 2 branch. You can take a look at the code for reference here:

https://github.com/vranki/ExtPlane-Panel/blob/feature/qml-ui/qmlui/panelitems/efis/Cdu.qml

Current ExtPlane version contain fixes that make using these datarefs possible. 

Thanks!

**Comment by waynepiekarski** [comment]
I was speaking to Philipp at FS Expo and he added the datarefs for me, but I just need to find some time to implement it.

Did you need to make new changes to the ExtPlane plugin to make this work? I wrote a test script a while ago and I was able to get all the datarefs out with the current ExtPlane plugin at the time and it worked fine for a test.

**Comment by vranki** [comment]
The text lines worked fine, but there were some issues with the binary style lines. That has now been fixed in version 1001. I now have working CDU display in ExtPlane-Panel including styling so you should be able to implement it also.

Thanks for talking to Philipp, this makes things much more easier for cockpit builders. Too bad FF A320 doesn't use the datarefs..

**Comment by waynepiekarski** [comment]
Ah great, thanks for the fixes, I hadn't gotten that far yet to notice it. I will start testing with your new plugin when I try to add support.

For the FF A320, it won't use the same datarefs, but WebFMC seems to support it, so it must be possible somehow?

**Comment by vranki** [comment]
Yeah, but I'd prefer FF doing changes in their product so it would work with any software using the new standard datarefs. I'll do a bug report about it in their ticket system.

**Comment by waynepiekarski** [comment]
Oh ok ... I've noticed every single aircraft does the datarefs completely differently. The new Laminar FMC is very different than the Zibo 737 and SSG 747 which I currently support. The FF aircraft are all from different developers too, and even simple knobs like the range on the EFIS have different named datarefs. There is no standard for anything except the flaps, parking brake, and some radios.

**Comment by vranki** [comment]
At least Laminar 737 and x737 use these datarefs. Not sure about Zibo. I hope all aircraft will switch to using these "standard" datarefs instead of their own. I created a support ticket to FF about this.

**Comment by jefken153** [comment]
Will there be an update to the ExtPlane plugin for the SSG B748 Anniv Edition V2.3.1 considering I get a blank screen in xplaneCDU. because it ask for an update.

![cdu xplane](https://user-images.githubusercontent.com/33124861/129480125-5f1e5ebc-dff1-4bee-ac88-d39f61b1eec0.jpg)

**Comment by waynepiekarski** [comment]
The ExtPlane plugin I publish on that link was an older version that I built myself with various fixes to the original ExtPlane. In the meantime, the plugin author merged my fixes and got their binary builds working again. We also merged in support for getting version info from ExtPlane, so getting ExtPlane from the original author is the best place to get this now. The older version still works, and that popup toast message is just a warning that you can ignore.
